### PR TITLE
Rewrite unit tests for utils module

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
|Related issue|
|---|
|#5060|

# Status update

## Test results
```
adriiiprodri@wazuh python3 -m coverage run --source=wazuh.utils -m pytest -x test_utils.py
=========================================================================== test session starts ===========================================================================
platform linux -- Python 3.7.5, pytest-5.4.1, py-1.7.0, pluggy-0.13.1
rootdir: /home/adriiiprodri/Desktop/git/wazuh/framework
plugins: html-2.0.1, mock-1.10.0, asyncio-0.12.0, aiohttp-0.3.0, cov-2.8.1, testinfra-5.0.0, tavern-1.0.0, metadata-1.8.0
collected 189 items                                                                                                                                                       

test_utils.py ..................................................................................................................................................... [ 78%]
........................................                                                                                                                            [100%]

=========================================================================== 189 passed in 0.51s ===========================================================================
```
```bash
adriiiprodri@wazuh coverage report -m
Name                              Stmts   Miss  Cover   Missing
---------------------------------------------------------------------------------------------
wazuh/framework/wazuh/utils.py     582      3    99%    750, 753, 789
```

### Lines uncovered
- Line 750 -> Method not implemented (`raise NotImplementedError`)
- Line 753 -> Method not implemented (`raise NotImplementedError`)
- Line 789 -> WazuhDB connection (`return WazuhDBConnection()`)

### CODE REVIEW

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 0341feac175d354deba0c8ee5578d32fb7905e14
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1989/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1989/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1989/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B1989.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B1989/Analysis_dynamic_valgrind-v4.0.0-B1989.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 549a53ae650e78f366409b23f11d7795d71b06a0
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2002/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2002/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2002/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2002.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2002/Analysis_dynamic_valgrind-v4.0.0-B2002.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** ac764f218b6696c600fbcc3e3a06bb00282676d2
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2333/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2333/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2333/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2333.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2333/Analysis_dynamic_valgrind-v4.0.0-B2333.log)

File: api/api/controllers/cluster_controller.py [hunk 1]
```diff
@@ -80,8 +80,9 @@ async def get_cluster_nodes(request, pretty=False, wait_for_complete=False, offs
                           wait_for_complete=wait_for_complete,
                           logger=logger,
                           local_client_arg='lc',
-                          rbac_permissions=request['token_info']['rbac_policies'],
-                          nodes=nodes
+                          rbac_permissions=get_permissions(connexion.request.headers['Authorization']),
```
Comment (Selutario) [inline api/api/controllers/cluster_controller.py:83]: Check this works correctly when [4300-dev-flask-rework-cluster-unittests ](https://github.com/wazuh/wazuh/pull/5095) is merged.

Comment (vicferpoy) [reply api/api/controllers/cluster_controller.py:83]:
```suggestion
                          rbac_permissions=request['token_info']['rbac_policies'],
```
We should use the `request` parameter instead.

File: api/api/controllers/cluster_controller.py [hunk 2]
```diff
@@ -109,8 +110,9 @@ async def get_healthcheck(request, pretty=False, wait_for_complete=False, list_n
...
```
Comment (vicferpoy) [inline api/api/controllers/cluster_controller.py:113]:
```suggestion
                          rbac_permissions=request['token_info']['rbac_policies'],
```
Same as above. Change this in every endpoint.

File: framework/wazuh/tests/test_utils.py [hunk 3]
```diff
@@ -123,8 +125,8 @@ def test_execute(mock_output):
 
 
 @pytest.mark.parametrize('error_effect, expected_exception', [
-    (CalledProcessError(returncode=10000, cmd='Unspected error', output='{"data":"Some data", "message":"Error", '
-                                                                        '"error":10000}'), 10000),
+    (CalledProcessError(returncode=1000, cmd='Unspected error', output='{"data":"Some data", "message":"Error", '
```
Comment (vicferpoy) [inline framework/wazuh/tests/test_utils.py:128]:
```suggestion
    (CalledProcessError(returncode=1000, cmd='Unexpected error', output='{"data":"Some data", "message":"Error", '
```

File: framework/wazuh/tests/test_utils.py [hunk 4]
```diff
@@ -123,8 +125,8 @@ def test_execute(mock_output):
...
+                                                                       '"error":1000}'), 1000),
     (Exception, 1002),
     (CalledProcessError(returncode=1, cmd='Unspected error', output={}), 1003),
     (CalledProcessError(returncode=1, cmd='Unspected error', output='{"error":10000}'), 1004),
```
Comment (vicferpoy) [inline framework/wazuh/tests/test_utils.py:132]:
```suggestion
    (CalledProcessError(returncode=1, cmd='Unexpected error', output={}), 1003),
    (CalledProcessError(returncode=1, cmd='Unexpectederror', output='{"error":10000}'), 1004),
```
Should `{"error":10000}` be `{"error":1000}`?

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** e45fefe55f1446c77c1cda598ad89f24aa7e651e
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2483/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2483/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2483/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2483.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2483/Analysis_dynamic_valgrind-v4.0.0-B2483.log)

**Pull request merged**